virtual-network/ipv6-vnet-with-two-vms/ipv6-vnet-with-two-vms.ps1

```diff
@@ -0,0 +1,154 @@
+-Name "dsPublicIP_v6" `
+-ResourceGroupName $rg.ResourceGroupName `
+-Location $rg.Location  `
+-AllocationMethod Dynamic `
+-IpAddressVersion IPv6
+
+$RdpPublicIP_1 = New-AzPublicIpAddress `
+-Name "RdpPublicIP_1" `
+-ResourceGroupName $rg.ResourceGroupName `
+-Location $rg.Location  `
+-AllocationMethod Dynamic `
+-IpAddressVersion IPv4
+
+  $RdpPublicIP_2 = New-AzPublicIpAddress `
+  -Name "RdpPublicIP_2" `
+  -ResourceGroupName $rg.ResourceGroupName `
+  -Location $rg.Location  `
+  -AllocationMethod Dynamic `
+  -IpAddressVersion IPv4
+
+
+#Create Basic load balancer
+$frontendIPv4 = New-AzLoadBalancerFrontendIpConfig `
+-Name "dsLbFrontEnd_v4" `
+-PublicIpAddress $PublicIP_v4
+
+$frontendIPv6 = New-AzLoadBalancerFrontendIpConfig `
+-Name "dsLbFrontEnd_v6" `
+-PublicIpAddress $PublicIP_v6
+
+$backendPoolv4 = New-AzLoadBalancerBackendAddressPoolConfig -Name "dsLbBackEndPool_v4"
+$backendPoolv6 = New-AzLoadBalancerBackendAddressPoolConfig -Name "dsLbBackEndPool_v6"
+
+$lbrule_v4 = New-AzLoadBalancerRuleConfig `
+-Name "dsLBrule_v4" `
+-FrontendIpConfiguration $frontendIPv4 `
+-BackendAddressPool $backendPoolv4 `
+-Protocol Tcp `
+-FrontendPort 80 `
+-BackendPort 80
+
+$lbrule_v6 = New-AzLoadBalancerRuleConfig `
+-Name "dsLBrule_v6" `
+-FrontendIpConfiguration $frontendIPv6 `
+-BackendAddressPool $backendPoolv6 `
+-Protocol Tcp `
+-FrontendPort 80 `
+-BackendPort 80
+
+$lb = New-AzLoadBalancer `
+-ResourceGroupName $rg.ResourceGroupName `
+-Location $rg.Location  `
+-Name "MyLoadBalancer" `
+-Sku "Basic" `
+-FrontendIpConfiguration $frontendIPv4,$frontendIPv6 `
+-BackendAddressPool $backendPoolv4,$backendPoolv6 `
+-LoadBalancingRule $lbrule_v4,$lbrule_v6
+
+#Create Availability Set
+$avset = New-AzAvailabilitySet `
+-ResourceGroupName $rg.ResourceGroupName `
+-Location $rg.Location  `
+-Name "dsAVset" `
+-PlatformFaultDomainCount 2 `
+-PlatformUpdateDomainCount 2 `
+-Sku aligned
+
+#Create Network Security Group and Rules
+$rule1 = New-AzNetworkSecurityRuleConfig `
+-Name 'myNetworkSecurityGroupRuleRDP' `
+-Description 'Allow RDP' `
+-Access Allow `
+-Protocol Tcp `
+-Direction Inbound `
+-Priority 100 `
+-SourceAddressPrefix * `
+-SourcePortRange * `
+-DestinationAddressPrefix * `
+-DestinationPortRange 3389
+
+$rule2 = New-AzNetworkSecurityRuleConfig `
+-Name 'myNetworkSecurityGroupRuleHTTP' `
+-Description 'Allow HTTP' `
+-Access Allow `
+-Protocol Tcp `
+-Direction Inbound `
+-Priority 200 `
+-SourceAddressPrefix * `
+-SourcePortRange 80 `
+-DestinationAddressPrefix * `
+-DestinationPortRange 80
+
+$nsg = New-AzNetworkSecurityGroup `
+-ResourceGroupName $rg.ResourceGroupName `
+-Location $rg.Location  `
+-Name "dsNSG1"  `
+-SecurityRules $rule1,$rule2
+
+#Create Virtual Network and Subnet
+# Create dual stack subnet config
+$subnet = New-AzVirtualNetworkSubnetConfig `
+-Name "dsSubnet" `
+-AddressPrefix "10.0.0.0/24","ace:cab:deca:deed::/64"
+
+# Create the virtual network
+$vnet = New-AzVirtualNetwork `
```

Comment:
Got an error here:
```
New-AzVirtualNetwork : Subscription .... is not registered for feature Microsoft.Network/AllowIPv6VirtualNetwork required to 
carry out the requested operation.
```

If there are additional configuration flags that need to be set to allow IPv6, they should either be called out as prerequisites in the comments or toggled by the script itself.

Reply:
I have added the registration information at the beginning of the script. The registration is to be only done once. Thanks!